core/internal/cltest/mocks.go

```diff
@@ -334,21 +334,26 @@ func fwdHeaders(actual, mock interface{}) {
 
 // MockSubscription a mock subscription
 type MockSubscription struct {
-	name    string
-	channel interface{}
-	Errors  chan error
+	name         string
+	channel      interface{}
+	unsubscribed bool
+	Errors       chan error
 }
 
 // EmptyMockSubscription return empty MockSubscription
-func EmptyMockSubscription() MockSubscription {
-	return MockSubscription{Errors: make(chan error, 1), channel: make(chan struct{})}
+func EmptyMockSubscription() *MockSubscription {
+	return &MockSubscription{Errors: make(chan error, 1), channel: make(chan struct{})}
 }
 
 // Err returns error channel from mes
-func (mes MockSubscription) Err() <-chan error { return mes.Errors }
+func (mes *MockSubscription) Err() <-chan error { return mes.Errors }
 
 // Unsubscribe closes the subscription
-func (mes MockSubscription) Unsubscribe() {
+func (mes *MockSubscription) Unsubscribe() {
+	if mes.unsubscribed {
+		return
+	}
+	mes.unsubscribed = true
```

Comment:
This looks suspiciously racy to me. Have you tried it with the race detector?

If `Unsubscribe` is called twice this may do something strange.